packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js

```diff
@@ -0,0 +1,299 @@
+   */
+  get errorMessage() {
+    return this._errorMessage;
+  }
+
+  /**
+   * Tells if there are more options to show in addition to the suggested options.
+   * @returns {boolean}
+   */
+  get isMoreOptionsVisible() {
+    return this.options.length > Math.max(Number(this.maxChoices), 1);
+  }
+
+  /**
+   * Whether the select input is visible.
+   * @type {boolean}
+   */
+  get isSelectVisible() {
+    return this._isSelectVisible || 0 >= this.suggestions.length;
+  }
+
+  /**
+   * Whether the suggested options are visible.
+   * @type{boolean}
+   */
+  get isSuggestionsVisible() {
+    return this._isSuggestionsVisible;
+  }
+
+  /**
+   * Returns the CSS class of the form.
+   * @returns {string}
+   */
+  get formClass() {
+    return `slds-form-element ${this.hasError ? 'slds-has-error' : ''}`;
+  }
+
+  /**
+   * Returns the options, excluding the suggestions.
+   * @returns {Array<object>}
+   */
+  get filteredOptions() {
+    return this.options.filter(
+      (option) =>
+        !this.classifications.some((item) => item.value === option.value)
+    );
+  }
+
+  /**
+   * Returns the suggestions to be shown.
+   * @returns {Array}
+   */
+  get suggestions() {
+    return this.classifications.slice(
+      0,
+      Math.max(Number(this.maxChoices), 1) - 1
+    );
+  }
+
+  /**
+   * Shows the select input.
+   * @returns {void}
+   */
+  showSelect() {
+    this._isSelectVisible = true;
+  }
+
+  /**
+   * Handles the selection of a suggestion.
+   * @returns {void}
+   */
+  handleSelectClassification(event) {
+    const suggestionId = event.target.dataset.suggestionId;
+    const value = event.target.value;
+    this.engine.dispatch(this.actions.logClassificationClick(suggestionId));
+    this.field.update(value);
+    this._errorMessage = '';
+    this._value = value;
+  }
+
+  /**
+   * Handles the select input change.
+   * @returns {void}
+   */
+  handleSelectChange(event) {
+    const value = event.target.value;
+    this.field.update(value);
+    this._errorMessage = '';
+    this._value = value;
+    if (this._isSuggestionsVisible && this.isMoreOptionsVisible) {
+      this._hideSuggestions();
+    }
+  }
+
+  /**
+   * Hide the suggested options.
+   * @returns {void}
+   */
+  _hideSuggestions() {
+    const suggestions = this.template.querySelectorAll('.slds-visual-picker');
+    suggestions.forEach((suggestion) => {
+      // @ts-ignore
+      suggestion.style.width = `${suggestion.clientWidth}px`;
+      suggestion.classList.add('visual-picker__hidden');
+    });
+  }
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {CaseAssistEngine} engine
+   */
+  initialize = (engine) => {
+    this.engine = engine;
+    this.field = CoveoHeadlessCaseAssist.buildCaseField(engine, {
+      options: {
+        field: `sf${this.fieldName.toLowerCase()}`,
```

Comment:
ouf, this is a discussion entirely...

I think it is wrong to assume that the field from the CaseAssist will always have this format. And as mentioned above on the `@api fieldName` property. You are playing with 3 different potential values here.

SF field API name => example `My_Custom_Field__c`

SF field label         => example `My Custom Field`

Coveo field name => example `sfmycustomfield` but can actually be ANYTHING the user wants. This is just the default value


I think I would do this:

```js
@api sfFieldAPIName;
@api coveoFieldName = `sf${this.sfFieldAPIName.toLowerCase()}`;
```

I am not even sure if we don't add a `c` at the end of the `sfmycustomfield` when it is a custom field in Salesforce
`sfmycustomfieldc`.

We used to by default, now I am not sure if it was removed.

Reply:
sfFieldAPIName and coveoFieldName added 👌